Logstash/logstash-output-applicationinsights/lib/logstash/outputs/applicationinsights.rb

```diff
@@ -0,0 +1,85 @@
+# encoding: utf-8
+require "logstash/outputs/base"
+require "logstash/namespace"
+
+require "application_insights"
+
+class LogStash::Outputs::ApplicationInsights < LogStash::Outputs::Base
+  config_name "applicationinsights"
+
+  config :ikey, :validate => :string, :required => true
+  config :dev_mode, :validate => :boolean, :required => false, :default => false
+  config :ai_message_field, :validate => :string, :required => false, :default => nil
+  config :ai_properties_field, :validate => :string, :required => false, :default => nil
+  config :ai_severity_level_field, :validate => :string, :required => false, :default => nil
+  config :ai_severity_level_mapping, :validate => :hash, :required => false, :default => nil
+
+  public
+  def register
+     create_client
+  end # def register
+
+  public
+  def multi_receive(events)
+    events.each do |event|
+      begin
+        ai_message = get_ai_message(event)
+        ai_properties = get_ai_properties(event)
+        ai_severity = get_ai_severity(event)
+
+        @client.track_trace(ai_message, ai_severity, { :properties => ai_properties })
+        @client.flush if @dev_mode
```

Comment:
Would it be better to flush after sending all events (i.e. just once per multi_receive invocation)?